pyccel/ast/builtin_objects/list_functions.py

```diff
@@ -0,0 +1,65 @@
+# coding: utf-8
+#------------------------------------------------------------------------------------------#
+# This file is part of Pyccel which is released under MIT License. See the LICENSE file or #
+# go to https://github.com/pyccel/pyccel/blob/master/LICENSE for full license details.     #
+#------------------------------------------------------------------------------------------#
+
+"""
+The List container has a number of built-in methods that are 
+always available.
+
+This module contains objects which describe these methods within Pyccel's AST.
+"""
+
+from pyccel.ast.internals import PyccelInternalFunction
+from pyccel.ast.datatypes import NativeHomogeneousList
+
+class ListPop(PyccelInternalFunction) :
+    """
+    Represents a call to the .pop() method.
+    
+    Represents a call to the .pop() method which
+    removes the item at the specified index. 
+    The method also returns the removed item.
+
+    Parameters
+    ----------
+    name : TypedAstNode
+        The name of the list.
+
+    index_elemnt : TypedAstNode
+        The current index value for the element to be popped.
+    """
+    __slots__ = ('_dtype','_precision', '_index','_list_variable')
+    _rank = 0
+    _order = None
+    _shape = None
+    _class_type = NativeHomogeneousList()
+    name = 'pop'
+    def __init__(self, name, index_elemnt=None):
+        if index_elemnt:
+            super().__init__(name, index_elemnt)
+        else:
+            super().__init__(name)
+        self._index = index_elemnt
+        self._list_variable = name
+        self._dtype = name.dtype
+        self._precision = name.precision
+
+    @property
+    def args(self):
+        """
+        The current index value for the element to be popped.
+
+        The current index value for the element to be popped.
+        """
```

Comment:
Can you explain why "args" is the best name to describe "The current index value for the element to be popped.". Why not something like `pop_index`?

The only information that I can get from seeing "args" in the code is that it contains more than 1 object (as args is plural), but this isn't true :/

Reply:
Fixed 61433ec7f31d1c86e7449b3b91482fae3adfc110